pkg/kube/controllers/extendedjob/job_reconciler.go

```diff
@@ -178,24 +180,28 @@ func (r *ReconcileJob) persistOutput(ctx context.Context, instance *batchv1.Job,
 		if err != nil {
 			return errors.Wrap(err, "getting pod output")
 		}
-		// Put full job output in debug log
-		ctxlog.Debug(ctx, string(result))
+
+		// Create secret
+		secretName := ejob.Spec.Output.NamePrefix + c.Name
 
 		var data map[string]string
 		err = json.Unmarshal(result, &data)
 		if err != nil {
+			ctxlog.WithEvent(&ejob, "ExtendedJob").Errorf(ctx, "non-JSON output was emitted for container '%s', secret '%s' cannot be created", instance.GetName(), secretName)
 			return errors.Wrap(err, "invalid output format")
 		}
 
-		// Create secret and persist the output
-		secretName := ejob.Spec.Output.NamePrefix + c.Name
+		// Put full job output in debug log
+		ctxlog.Debugf(ctx, "Printing logs from container %s: %s", c.Name, string(result))
```

Comment:
We talked about that in the daily, can you remove that, while you're at it? Sorry :)